src/ngraph/frontend/fluid/operators/matmul_pd.cpp

```diff
@@ -0,0 +1,138 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <numeric>
+
+#include "matmul_pd.hpp"
+#include "ngraph/builder/matmul_factory.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/op/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo fluid::MatMulPd::type_info;
+fluid::MatMulPd::MatMulPd(const Output<Node>& A,
+                   const Output<Node>& B,
+                   const bool& transpose_a,
+                   const bool& transpose_b)
+    : FusedOp(OutputVector{A, B})
+    , m_transpose_a{transpose_a}
+    , m_transpose_b{transpose_b}
+{
+    constructor_validate_and_infer_types();
+}
+
+template<class Input>
+void DecomposeLogic(Input& input, bool transpose, bool reverse=false)
+{
+    auto _rank = input.get_shape().size();
+    if(_rank < 2)
+    {
+        if(_rank)
+        {            
+            if(reverse)
+            {
+               input = make_shared<op::Reshape>(input, AxisVector{0},Shape{input.get_shape()[0],1});
+            }
+            else
+            {
+               input = make_shared<op::Reshape>(input, AxisVector{0},Shape{1,input.get_shape()[0]});
+            }
+        }
+        else
+        {
+            input = make_shared<op::Reshape>(input,AxisVector{},Shape{1, 1});
+        }              
+    _rank=2;     
+    }
+    if (transpose)
+    {
+        vector<size_t> _axes_order(_rank);
+        iota(_axes_order.begin(), _axes_order.end(), 0);
+        swap(_axes_order[_rank - 1], _axes_order[_rank - 2]);
+        input = builder::reorder_axes(input, _axes_order);
+    }
+}
+
+inline NodeVector remove_1(std::shared_ptr<ngraph::Node> input_node)
+{
+    auto _input_shape = input_node->get_shape();
+    AxisVector _axis( _input_shape.size() );
+    iota(_axis.begin(),_axis.end(),0);
+    Shape _shape(_input_shape.begin(),_input_shape.end());
+    auto _b_remove = std::remove(_shape.begin(),_shape.end(),1);
+    _shape.erase(_b_remove,_shape.end());
+    Output<Node> _node( input_node );  
+    auto _reshape = make_shared<op::Reshape>( _node , _axis, _shape);
+    NodeVector _final_vector{ _reshape };
+    return _final_vector; 
+}
+
+ void fluid::MatMulPd::pre_validate_and_infer_types()
+{
+    element::Type input_element_type = get_input_element_type(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          input_element_type.is_dynamic() || input_element_type.is_real(),
+                          "Argument element type must be f16, bf16, f32, f64 or dynamic (got ",
+                          input_element_type,
+                          ").");
+    set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+
+    if (is_dynamic())
+    {
+        return;
+    }
+}
+NodeVector fluid::MatMulPd::decompose_op() const
+{
+     auto _A = input_value(0);
```

Comment:
do we want to use those underline prefix?

Reply:
done